src/axolotl/utils/config/models/input/v0_4_1/__init__.py

```diff
@@ -1548,3 +1550,21 @@ def check_multigpu_unsloth(cls, data):
                     "unsloth_lora_mlp, unsloth_lora_qkv, and unsloth_lora_o are not compatible with multi-GPU training."
                 )
         return data
+
+    @model_validator(mode="before")
+    @classmethod
+    def check_adopt_torch_version(cls, data):
+        if (data.get("optimizer") is not None) and ("adopt" in data.get("optimizer")):
+            env_capabilities = data.get("env_capabilities")
+            torch_version = env_capabilities.get("torch_version")
+
+            if torch_version is None:
+                import torch
+
+                torch_version = str(torch.__version__).split("+", maxsplit=1)[0]
+
+            if version.parse(torch_version) < version.parse("2.5.1"):
+                raise ValueError(
+                    "ADOPT optimizer is incompatible with torch version < 2.5.1"
+                )
```

Comment:
above in `utils` you created a new `is_torch_min` helper function, but never used it. I assumed you intended to use it here?

Reply:
that was my original intention but since we are using env_capabilities `is_torch_min` becomes unnecessary, tho I left it in `utils` cuz I thought it might be useful for some other functionalities in the future. Should I remove it from `utils`?

Reply:
yes, let's remove `is_torch_min` then please as it's unused code then.

Reply:
Got it, I just removed it.